neuro_san/test/assessor/assessor_assert_forwarder.py

```diff
@@ -42,199 +42,48 @@ def get_fail_dicts(self) -> List[Dict[str, Any]]:
         """
         return self.fail
 
-    def assertTrue(self, expr: Any, msg: str = None):
+    # pylint: disable=invalid-name
+    def assertGist(self, gist: bool, acceptance_criteria: str, text_sample: str, msg: str = None):
```

Comment:
You will see that in the AssertForwarder interface there is now a special assertGist().
This implementation uses that to collect data from the GistAgentEvaluator without having to hackily parse assert message and without hackily assuming assertTrue() and assertFalse() are all about gist, when they are not necessarily.